// Промисы: сравните then и catch
/* Являются ли фрагменты кода ниже эквивалентными? 
Другими словами, ведут ли они себя одинаково во всех обстоятельствах, 
для всех переданных им обработчиков?
*/
promise.then(f1).catch(f2); // если произойдет ошибка в f1, то catch ее обработает

// Против:
promise.then(f1, f2); // ошибка произошедшая в f1 не будет обработана